### ISSUE TITLE
I used the yaml model I built and loaded the pre-trained weights I got earlier. Why didn't the weight sense work

### ISSUE THREAD

**Issue by Sparklexa** [ultralytics/ultralytics#7356]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I used the yaml model I built and loaded the pre-trained weights I got earlier. Why didn't the weight sense work.
But if I didn't use my own yaml file and just used the pre-training weights, the accuracy would be above 90 in the first epoch.

### Additional

This is code that uses pre-trained weights：
```
from ultralytics import YOLO
# Load a model
model = YOLO('module/yolov8-SPD-Conv.yaml')
model = YOLO('runs/detect/trainbest/weights/best.pt')
model = YOLO('module/yolov8-SPD-Conv.yaml').load('runs/detect/trainbest/weights/best.pt')
results = model.train(data='data/balls.yaml', batch=8, epochs=50, imgsz=640,
                      device=3, workers=8, resume=False, label_smoothing=0.1, close_mosaic=10, pretrained=True)
```
This is the result of the first round of training：
![image](https://github.com/ultralytics/ultralytics/assets/150432487/37e858bf-89eb-4071-bf81-0a6ef54cb50d)

This is code that does not use pre-training weights：
```
from ultralytics import YOLO
# Load a model
model = YOLO('module/yolov8-SPD-Conv.yaml')
results = model.train(data='data/balls.yaml', batch=8, epochs=50, imgsz=640,
                      device=3, workers=8, resume=False, label_smoothing=0.1, close_mosaic=10, pretrained=Flase)
```
This is the result of the first round of training：
![image](https://github.com/ultralytics/ultralytics/assets/150432487/414bb1f2-c4b6-48f8-a0cb-d6e36e89c9e2)

**Comment by glenn-jocher** [comment]
@Sparklexa hello! It seems like there might be a mismatch between your custom YAML model configuration and the pre-trained weights you're trying to use. When you use pre-trained weights, it's important that the model architecture defined in your YAML file matches the architecture the weights were trained on. If there's a discrepancy, the weights may not be applied correctly, leading to unexpected results.

When you don't use your own YAML and rely on the pre-trained weights alone, you're likely using a default model configuration that matches the pre-trained weights, hence the high accuracy from the start.

Please double-check your custom YAML file to ensure it aligns with the pre-trained model's architecture. If you need further guidance on model configurations and using pre-trained weights, our documentation at https://docs.ultralytics.com can be a helpful resource. 😊

**Comment by zeobec** [comment]
Hi @Sparklexa Have you successfully implemented SPD-Conv into YOLOv8?
